source/Calamari.Shared/Integration/Packages/NuGet/NuGetV3Downloader.cs

```diff
@@ -8,23 +8,71 @@
 using System.Linq;
 using System.Net;
 using System.Net.Http;
+using System.Web;
 using Calamari.Common.Commands;
 using Calamari.Common.Plumbing.Logging;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Octopus.CoreUtilities.Extensions;
 using Octopus.Versioning;
+using Octopus.Versioning.Semver;
 
 namespace Calamari.Integration.Packages.NuGet
 {
     internal static class NuGetV3Downloader
     {
+        public static bool CanHandle(Uri feedUri, ICredentials feedCredentials, TimeSpan httpTimeout)
+        {
+            if (feedUri.ToString().EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+            
+            return IsJsonEndpoint(feedUri, feedCredentials, httpTimeout);
+        }
+
+        static bool IsJsonEndpoint(Uri feedUri, ICredentials feedCredentials, TimeSpan httpTimeout)
+        {
```

Comment:
It feels like we've duplicated most of the code from `GetHttp` below. Is there a reasonable way to avoid this duplication?